### PR TITLE
fix(calendar): 🐛  find a way to use utilities for data attributes

### PR DESCRIPTION
**Pull request by navin-moorthy**

### CODE REVIEW

File: src/calendar/Calendar.tsx [hunk 1]
```diff
@@ -31,6 +34,22 @@ export const Calendar: React.FC<CalendarProps> = props => {
...
+
+  return (
+    <CalendarProvider value={context}>
+      <RenderlessCalendar {...ctx} className={theme.calendar.base}>
+        {children}
+      </RenderlessCalendar>
+    </CalendarProvider>
+  );
+};
+
```
Comment (navin-moorthy) [inline src/calendar/Calendar.tsx:52]: Creating a range calendar couldn't be so easier 😆

File: src/calendar/stories/Calendar.stories.tsx [hunk 2]
```diff
@@ -63,3 +64,38 @@ const Template: Story = () => {
 
 export const Default = Template.bind({});
 Default.args = {};
+
+const RangeTemplate: Story = () => {
+  return (
+    <RangeCalendar>
```
Comment (navin-moorthy) [inline src/calendar/stories/Calendar.stories.tsx:70]: Users just need to do this, they get normal/Range

```suggestion
    <Calendar>
```

File: src/calendar/stories/calendar.css [hunk 3]
```diff
@@ -4,40 +4,12 @@
...
-  .calendar .calendar__cell[data-is-selection-end] {
-    @apply bg-blue-500 rounded-r-lg text-white !important;
-  }
-
-  .calendar .calendar__cell[data-is-range-selection]:focus-within {
+  /** Need to add the variant for our data variant via addVariant plugin */
+  /** Need to fix the tailwind override issue with focuswith with the new variants */
+  /** Tailwind override detects only the plain css doesn't support variants */
+  /** Or Look into styling the calendar like the storybook example from renderless component */
+  /* .calendar .calendar__cell[data-is-range-selection]:focus-within {
```
Comment (navin-moorthy) [inline src/calendar/stories/calendar.css:12]: Things that needs to be solved later on when working on Calendar after design is done !

File: tailwind.config.js [hunk 4]
```diff
@@ -43,5 +61,50 @@ module.exports = {
...
+    }),
+    plugin(function ({ addVariant, e }) {
+      addVariant("is-range-end", ({ modifySelectors, separator }) => {
+        modifySelectors(({ className }) => {
+          return `.${e(
+            `is-range-end${separator}${className}`,
+          )}[data-is-selection-end]`;
+        });
+      });
+    }),
```
Comment (navin-moorthy) [inline tailwind.config.js:108]: Of course, we can simplify this into a single plugin !
Left it separately for experimenting, we can simplify them at last IMO

**Review by anuraghazra — APPROVED** [review]

**Pull request merged**